src/inspect_ai/util/_sandbox/docker/docker.py

```diff
@@ -423,10 +424,21 @@ async def connection(self) -> SandboxConnection:
         if container:
             return SandboxConnection(
                 type="docker",
-                command=f"docker exec -it {container} bash -l",
+                command=shlex.join(
+                    [
+                        "docker",
+                        "exec",
+                        "-it",
+                        *(["--user", user] if user else []),
+                        container,
+                        "bash",
+                        "-l",
+                    ]
+                ),
                 vscode_command=[
                     "remote-containers.attachToRunningContainer",
                     container,
+                    # TODO: Is there a way to pass the user here?
```

Comment:
I'm not sure if there is. If not, maybe we shouldn't show the "open in VS Code" links in Inspect's output if `user` is not None. The links could cause the human baseliner to access the environment as the wrong user.

Reply:
Yes, in this case I think we should hide the VS Code command link.